sklearn/preprocessing/data.py

```diff
@@ -2579,6 +2579,151 @@ def quantile_transform(X, axis=0, n_quantiles=1000,
                          " axis={}".format(axis))
 
 
+class BoxCoxTransformer(BaseEstimator, TransformerMixin):
+    """Apply the Box-Cox transform featurewise to make data more Gaussian-like.
+
+    The Box-Cox transformation is a monotonic transformation that is
+    applied to make data more Gaussian-like. This is useful for solving
+    modeling issues related to heteroscedasticity (non-constant variance),
+    or other situations where normality is desired.
+
+    Box-Cox requires input data to be strictly positive. The optimal value
+    for the parameter, lambda, is estimated through maximum likelihood.
+
+    The Box-Cox transformation is given by::
+
+        if lambda == 0:
+            X_trans = log(X)
+        else:
+            X_trans = (X ** lambda - 1) / lambda
+
+    Parameters
+    ----------
+    copy : boolean, optional, default=True
+        Set to False to perform inplace computation.
+
+    Attributes
+    ----------
+    lambdas_ : array of float, shape (n_features,)
+        The parameters of the Box-Cox transformation for the selected features.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.preprocessing import BoxCoxTransformer
+    >>> boxcox = BoxCoxTransformer()
+    >>> data = [[1, 2], [3, 2], [4, 5]]
+    >>> print(boxcox.fit(data))
+    BoxCoxTransformer(copy=True)
+    >>> print(np.round(boxcox.lambdas_, 3))
+    [ 1.052 -2.345]
+    >>> print(np.round(boxcox.transform(data), 3))
+    [[ 0.     0.342]
+     [ 2.068  0.342]
+     [ 3.135  0.417]]
+
+    Notes
+    -----
+    For a comparison of the different scalers, transformers, and normalizers,
+    see :ref:`examples/preprocessing/plot_all_scaling.py
+    <sphx_glr_auto_examples_preprocessing_plot_all_scaling.py>`.
+
+    References
+    ----------
+    G.E.P. Box and D.R. Cox, "An Analysis of Transformations", Journal of the
+    Royal Statistical Society B, 26, 211-252 (1964).
+
+    """
+    def __init__(self, copy=True):
+        self.copy = copy
+
+    def fit(self, X, y=None):
+        """Estimate the optimal lambda for each feature using MLE.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The data used to estimate the Box-Cox transformation parameters.
+
+        y : Ignored
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+
+        if np.any(X <= 0):
+            raise ValueError("The Box-Cox transformation can only be applied "
+                             "to strictly positive data")
+
+        self.lambdas_ = []
+        for col in X.T:
+            _, lmbda = stats.boxcox(col, lmbda=None)
+            self.lambdas_.append(lmbda)
```

Comment:
you have promised an array, but provide a list. please make it an array, and test that it is so.

Reply:
Done